provd/cmd/provd/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"log/slog"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/canonical/ubuntu-desktop-provision/provd/cmd/provd/daemon"
+)
+
+func main() {
+	a := daemon.New()
+	os.Exit(run(a))
+}
+
+type app interface {
+	Run() error
+	UsageError() bool
+	Hup() bool
+	Quit()
+}
+
+func run(a app) int {
+	defer installSignalHandler(a)()
+
+	if err := a.Run(); err != nil {
+		slog.Error("applicatoin run failed", "errorDetail", err)
```

Comment:
```suggestion
		slog.Error("application run failed", "errorDetail", err)
```

But look at my other comments about slog :)